mage_ai/data_preparation/models/pipeline.py

```diff
@@ -58,9 +60,32 @@ def get_all_pipelines(self, repo_path):
             os.mkdir(pipelines_folder)
         return os.listdir(os.path.join(repo_path, PIPELINES_FOLDER))
 
-    def execute(self):
-        # TODO: implement execution logic
-        pass
+    # async function for parallel processing
+    async def execute(self):
+        tasks = dict()
+        blocks = Queue()
+        for b in self.block_configs:
```

Comment:
I guess you reload the config because self.block_configs is not updated. you can use self.blocks_by_uuid.values() to get updated blocks.